src/m365/spo/commands/site/site-admin-remove.ts

```diff
@@ -0,0 +1,289 @@
+    return responseContent.value;
+  }
+
+  private async getCorrectLoginName(options: Options): Promise<string> {
+    if (options.userId || options.userName) {
+      let requestUrl: string = `https://graph.microsoft.com/v1.0/users/${options.userId}`;
+
+      if (options.userName) {
+        requestUrl = `https://graph.microsoft.com/v1.0/users('${formatting.encodeQueryParameter(options.userName)}')`;
+      }
+
+      const requestOptions: CliRequestOptions = {
+        url: requestUrl,
+        headers: {
+          accept: 'application/json;odata.metadata=none'
+        },
+        responseType: 'json'
+      };
+
+      const user = await request.get<any>(requestOptions);
+      if (user && user.userPrincipalName) {
+        return `i:0#.f|membership|${user.userPrincipalName}`;
+      }
+
+      throw 'User not found.';
+    }
+    else {
+      const group = options.groupId ? await entraGroup.getGroupById(options.groupId) : await entraGroup.getGroupByDisplayName(options.groupName!);
+      //for entra groups, M365 groups have an associated email and security groups don't
+      if (group?.mail) {
+        //M365 group is prefixed with c:0o.c|federateddirectoryclaimprovider
+        return `c:0o.c|federateddirectoryclaimprovider|${group.id}`;
+      }
+      else {
+        //security group is prefixed with c:0t.c|tenant
+        return `c:0t.c|tenant|${group?.id}`;
+      }
+    }
+  }
+
+  private async setSiteAdminsAsAdmin(adminUrl: string, siteId: string, admins: AdminUserResult[]): Promise<void> {
+    const requestOptions: CliRequestOptions = {
+      url: `${adminUrl}/_api/SPOInternalUseOnly.Tenant/SetSiteSecondaryAdministrators`,
+      headers: {
+        accept: 'application/json;odata=nometadata',
+        'content-type': 'application/json;charset=utf-8'
+      },
+      data: {
+        secondaryAdministratorsFieldsData: {
+          siteId: siteId,
+          secondaryAdministratorLoginNames:
+            admins.map(u => u.loginName)
+        }
+      }
+    };
+
+    await request.post<string>(requestOptions);
+  }
+
+  private async callAction(logger: Logger, args: CommandArgs, loginNameToRemove: string): Promise<void> {
+    await logger.logToStderr('Removing site administrator...');
+
+    const primaryOwnerLogin = await this.getPrimaryOwnerLoginFromSite(args.options.siteUrl);
+    if (loginNameToRemove === primaryOwnerLogin) {
+      throw 'You cannot remove the primary site collection administrator.';
+    }
+
+    const requestOptions: CliRequestOptions = {
+      url: `${args.options.siteUrl}/_api/web/siteusers('${formatting.encodeQueryParameter(loginNameToRemove)}')`,
+      headers: {
+        'accept': 'application/json',
+        'X-Http-Method': 'MERGE',
+        'If-Match': '*'
+      },
+      data: { IsSiteAdmin: false },
+      responseType: 'json'
+    };
+    await request.post(requestOptions);
+  }
+
+  private async getPrimaryOwnerLoginFromSite(siteUrl: string): Promise<string> {
```

Comment:
also this we already have in the `site admin list`.
Lets extract this to a new `util` in `spo` and resue it in both places 👍

Reply:
Done